libzstd/encoder/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+use zstd::stream::Encoder;
+use zstd::zstd_safe::{CParameter, ParamSwitch};
+
+// re-export zstd
+pub use zstd;
+
+// we use offset window no more than = 17
+// TODO: use for multi-block zstd.
+#[allow(dead_code)]
+pub const CL_WINDOW_LIMIT: usize = 17;
+
+/// zstd block size target.
+pub const N_BLOCK_SIZE_TARGET: u32 = 124 * 1024;
+
+/// Maximum number of blocks that we can expect in the encoded data.
+pub const N_MAX_BLOCKS: u64 = 10;
+
+/// Zstd encoder configuration
```

Comment:
a quick question: the `encoder` is another crate? it isn't a part of libstd?

Reply:
since `aggregator` has to depend on this crate. It is not suitable to make it a dynamic lib.

So we provide `encoder` for `aggregator`, then wrap `encoder` (a static lib) into libzstd (a dynamic lib for Golang side)

Reply:
got it